### ISSUE TITLE
Fix S4261 RSPEC: Rule is based off of errant examples and not the text rule. 

### ISSUE THREAD

**Issue by StingyJack** [SonarSource/sonar-dotnet#3928]
Both the compliant and noncompliant examples are missing the async modifier for the function. That is what determines if the method should be suffixed with Async or not.

current compliant example
``` csharp
public class Foo
  {
    public Task ReadAsync(byte [] buffer, int offset, int count, CancellationToken cancellationToken)
  }
```
what it should be
``` csharp
public class Foo
  {
    public async Task ReadAsync(byte [] buffer, int offset, int count, CancellationToken cancellationToken) {}
  }
```
… though even this is strange as the method name is Read but this doesnt return the result of reading anything.

What looks like is that this rule has referenced the example code that is incorrect, and not the [section](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap#naming-parameters-and-return-types) that states the guidance.

> Asynchronous methods in TAP include the `Async` suffix after the operation name for methods that return awaitable types, such as [Task](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task), [Task<TResult>](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task-1), [ValueTask](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask), and [ValueTask<TResult>](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1). For example, an asynchronous `Get` operation that returns a `Task<String>` can be named `GetAsync`

Also posted at https://community.sonarsource.com/t/s4261-has-incorrect-examples/36324/3

**Comment by StingyJack** [comment]
For the record, this GH issue documents how the advice from MSFT has changed. https://github.com/dotnet/docs/issues/23834
